cli/flox/src/commands/init/node.rs

```diff
@@ -309,20 +311,7 @@ impl Node {
             version = node.as_ref().unwrap_or(&"unspecified".to_string()),
             "trying to find compatible node version"
         );
-        // Prioritize newer major versions by putting them in order. The bare "nodejs"
-        // packages goes last because an old nodejs version that matches against it will
-        // pin the package group to an old page.
-        let nodejs_packages = [
-            "nodejs_23",
-            "nodejs_22",
-            "nodejs_21",
-            "nodejs_20",
-            "nodejs_19",
-            "nodejs_18",
-            "nodejs_16",
-            "nodejs_14",
-            "nodejs",
-        ];
+        let nodejs_packages = Self::get_available_node_packages(flox).await?;
```

Comment:
I checked with @billlevine and the order of the package groups in the response are preserved from the order in the request (I just realized I deleted my comment explaining this, I'll add it back in before merging). This ensures that when we create the package groups with the highest priority (e.g. latest) version first, we can take the first resolved group from the response.

Reply:
Docs and test:

- https://github.com/flox/catalog-server/issues/263